src/ArmTemplates/Commands/Executors/ExtractorExecutor.cs

```diff
@@ -68,76 +78,52 @@ public class ExtractorExecutor
 
-            if (splitAPIs)
+            if (this.extractorParameters.SplitApis)
             {
-                // create split api templates for all apis in the sourceApim
-                await this.GenerateSplitAPITemplates(fileNames);
-                await this.GenerateTemplates(new ExtractorParameters(this.extractorConfig), null, null, fileNames, null);
+                Logger.LogInformation("Starting templates with splitting for each API extraction...");
+                await this.GenerateSplitAPITemplates();
+                await this.GenerateTemplates(this.extractorParameters.FilesGenerationRootDirectory, null, null, null);
             }
-            else if (hasVersionSetName)
+            else if (!string.IsNullOrEmpty(this.extractorParameters.ApiVersionSetName))
             {
-                // create split api templates and aggregated api templates for this apiversionset
-                await this.GenerateAPIVersionSetTemplates(this.extractorConfig, fileNames);
+                Logger.LogInformation("Starting API version set templates extraction...");
+                await this.GenerateAPIVersionSetTemplates();
             }
-            else if (hasMultipleAPIs)
+            else if (!this.extractorParameters.MultipleApiNames.IsNullOrEmpty())
             {
-                // generate templates for multiple APIs
-                await this.GenerateMultipleAPIsTemplates(this.extractorConfig, fileNames);
+                Logger.LogInformation("Launching multiple APIs templates extraction...");
+                await this.GenerateMultipleAPIsTemplates();
             }
-            else if (hasSingleApi && includeRevisions)
+            else if (!string.IsNullOrEmpty(this.extractorParameters.SingleApiName) && this.extractorParameters.IncludeAllRevisions)
             {
-                // handle single API include Revision extraction
-                await this.GenerateSingleAPIWithRevisionsTemplates(this.extractorConfig, singleApiName, fileNames);
+                Logger.LogInformation("Launching single API with revisions templates extraction...");
+                await this.GenerateSingleAPIWithRevisionsTemplates();
             }
             else
             {
-                // create single api template or create aggregated api templates for all apis within the sourceApim
-                if (hasSingleApi)
-                {
-                    Logger.LogInformation("Executing extraction for {0} API ...", singleApiName);
-                }
-                else
-                {
-                    Logger.LogInformation("Executing full extraction ...");
-                }
-
-                await this.GenerateTemplates(new ExtractorParameters(this.extractorConfig), singleApiName, null, fileNames, null);
+                Logger.LogInformation("No specific parameters are set, executing extraction for {0} API...", this.extractorParameters.SingleApiName);
```

Comment:
`this.extractorParameters.SingleApiName` can be null so this log message can make no sense. Why not keep the 2 different messages depending on whether or not it's null?